src/Sylius/Bundle/WebBundle/Resources/views/Frontend/Product/_singleBox.html.twig

```diff
@@ -1,8 +1,9 @@
+{% from 'SyliusWebBundle:Frontend/Product:macros.html.twig' import sylius_display_price_with_original %}
 <div class="product-box">
     <a href="{{ path(product) }}"><h4>{{ product.name|truncate(18) }}</h4></a>
     <a href="{{ path(product) }}">
         <img class="img-rounded img-responsive" src="{{ product.image ? product.image.path|imagine_filter('sylius_medium') : 'http://placehold.it/262x255' }}" alt="{{ product.name }}" />
-        <span class="label label-primary">{{ sylius_calculate_price(product.masterVariant)|sylius_price }}</span>
```

Comment:
According to your error, you do not call `sylius_calculate_price()` in your macro. Moreover, current `isReduced` method in `ProductVariant` isn't aware of price calculator, so those channel differences are not processed (and it may introduce some errors if for some channel price is higher than regular price).